### DISCUSSION TITLE
A way to open specific urls (from anywhere) to Firefox PWA

### DISCUSSION THREAD

**Discussion by jeannnemelanie** [filips123/PWAsForFirefox#252] [Q&A]
Hi 👋

First, thanks for Firefox PWA! I'm really happy to be able to create Firefox PWAs 🚀

I use [Finicky](https://github.com/johnste/finicky) to be able to open different urls in different browers :
> Finicky is a macOS application that allows you to set up rules that decide which browser is opened for every link or url. With Finicky as your default browser, you can tell it to open Facebook or Reddit in one browser, and Trello or LinkedIn in another.

There is a special configuration to open Chrome & Brave PWA with Finicky. It work really well. What we have to do is to know app ID and Profile. I tried to use the same sort of config for Firefor PWA. It opens Firefox but it do not open the related PWA. Would you suggest another config to be able to open Firefox PWA ? 👉 https://github.com/johnste/finicky/wiki/Troubleshooting-&-Tips#i-want-to-use-a-chrome-app-shortcut-for-a-certain-website

I was thinking of hacking Finicky with running bash command instead of Finicky rules by using le `launch` command. But when a Firefox PWA is already open, the `launch` command opens another PWA window for the same app. Wich is not ideal.

**Comment by filips123** [comment]
What happens if you set Finicky's `browser` field to the path of PWA bundle for your PWA (located in `~/Applications`)? Maybe this could work.

> But when a Firefox PWA is already open, the `launch` command opens another PWA window for the same app.

What do you expect to happen if the same PWA is already opened but you click on another link to it? If it just replaced its content with the new load, you could potentially lose your original progress or stuff you entered in.

**Comment by jeannnemelanie** [comment]
> What happens if you set Finicky's browser field to the path of PWA bundle for your PWA (located in ~/Applications)? Maybe this could work.

It works but it opens 2 windows of the same PWA (with only 1 app icon shown as opened in the menubar). But then, when I click a second time in the same link (outside PWA & Firefox, for example in a note document). It opens another PWA app icon in the menubar without really openning the PWA window or updating the first PWA opened.

```
{
      match: "*.instagram.com*",
      browser: "/Users/my_name/Applications/Instagram.app"
}
```

> What do you expect to happen if the same PWA is already opened but you click on another link to it? If it just replaced its content with the new load, you could potentially lose your original progress or stuff you entered in.

If I set-up my PWA with tabs I would like the url to be opened in a new tab. Otherwise it's ok to open another window of the same PWA app to avoid losing anything already in used. Or maybe this could be a setting of the PWA so you can choose different behavior for different PWA (new tab, new window, current/main window).

**Reply by filips123** [reply]
> It works but it opens 2 windows of the same PWA (with only 1 app icon shown as opened in the menubar).

I will try to check why this happens and fix it.

> If I set-up my PWA with tabs I would like the url to be opened in a new tab. 

There is already "Open a web app in an existing window of that web app" in the PWA settings (available only if the tabs mode is enabled). Please check if it works when opening from Finicky and let me know.

**Reply by jeannnemelanie** [reply]
> There is already "Open a web app in an existing window of that web app" in the PWA settings (available only if the tabs mode is enabled). Please check if it works when opening from Finicky and let me know.

Oh I did not notice this setting, sorry about that! I just tested it and it is not working all the time. Sometimes it works as expected (open new link in a new tab) but sometimes it "opens" a new icon in the dock, this icon is jumping but nothing much is working (no new tab in the opened PWA, no new PWA window either). If I force close the new icon in the dock, then links open in the main PWA window in a new tab.

> I will try to check why this happens and fix it.

Thank you 🙏  I just realized when I used the "menubar" word in my previous message (13th december) I was actually meaning "dock". Sorry!
